### PR TITLE
[#9304] honor event type in query

### LINKED ISSUE #9304: DEMIS and Infra sync dates are mixed up
### Bug Description

For synchronizing lab messages and  infrastructure data, we rely on the  `SystemEventFacadeEjb`. However, `getLatestSuccessByType` does ignore the type which causes the query to return an arbitrarily typed sync event, causing lab message and infra sync to clash.

### Steps to Reproduce

Checkout the current S2S Testbed and test DEMIS sync with `sormas_b`

### Expected Behavior

`getLatestSuccessByType` honers the type of the system event in the query.

### PR DESCRIPTION
**Pull request by JonasCir**
Fixes #9304 
@FredrikSchaeferVitagroup Same commit, but I had to change the branches

### CODE REVIEW

**Comment by JonasCir** [comment]
@MateStrysewske Just FYI

Once this is merged, you can plan the release :)

File: sormas-backend/src/main/java/de/symeda/sormas/backend/systemevent/SystemEventFacadeEjb.java [hunk 1]
```diff
@@ -56,21 +58,20 @@ public SystemEventDto getLatestSuccessByType(SystemEventType type) {
 		CriteriaQuery<SystemEvent> cq = cb.createQuery(SystemEvent.class);
 		Root<SystemEvent> systemEventRoot = cq.from(SystemEvent.class);
 
-		cq.where(cb.equal(systemEventRoot.get(SystemEvent.STATUS), SystemEventStatus.SUCCESS));
+		final Predicate equalSystemEventStatus = cb.equal(systemEventRoot.get(SystemEvent.STATUS), SystemEventStatus.SUCCESS);
```
Comment (StefanKock) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/systemevent/SystemEventFacadeEjb.java:61]: I would not assign these as method variables, but directly inline them into the WHERE method.

File: sormas-backend/src/main/java/de/symeda/sormas/backend/systemevent/SystemEventFacadeEjb.java [hunk 2]
```diff
@@ -121,7 +122,6 @@ public SystemEvent fromDto(@NotNull SystemEventDto source, SystemEvent target, b
 	}
 
 	public SystemEventDto toDto(SystemEvent source) {
-
```
Comment (StefanKock) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/systemevent/SystemEventFacadeEjb.java:124]: This formatting violates [formatting guidelines](https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/docs/CONTRIBUTING.md#development-contributing-guidelines)
> When defining a method, enter a blank line before starting to write its body (except for methods with only one line of code, e.g. most getters and setters).

File: sormas-backend/src/main/java/de/symeda/sormas/backend/systemevent/SystemEventFacadeEjb.java [hunk 3]
```diff
@@ -150,7 +150,6 @@ public void deleteAllDeletableSystemEvents(int daysAfterSystemEventGetsDeleted)
 	}
 
 	public void deleteAllDeletableSystemEvents(LocalDateTime notChangedUntil) {
-
```
Comment (StefanKock) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/systemevent/SystemEventFacadeEjb.java:153]: This formatting violates [formatting guidelines](https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/docs/CONTRIBUTING.md#development-contributing-guidelines)
> When defining a method, enter a blank line before starting to write its body (except for methods with only one line of code, e.g. most getters and setters).

File: sormas-backend/src/main/java/de/symeda/sormas/backend/systemevent/SystemEventFacadeEjb.java [hunk 4]
```diff
@@ -56,21 +58,20 @@ public SystemEventDto getLatestSuccessByType(SystemEventType type) {
...
 
 		return QueryHelper.getFirstResult(em, cq, this::toDto);
 	}
 
-
 	@Override
 	@TransactionAttribute(TransactionAttributeType.REQUIRES_NEW)
 	public void saveSystemEvent(@Valid SystemEventDto dto) {
 		SystemEvent systemEvent = systemEventService.getByUuid(dto.getUuid());
-
```
Comment (StefanKock) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/systemevent/SystemEventFacadeEjb.java:70]: If reformatting, please add empty line directly after method.

[formatting guidelines](https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/docs/CONTRIBUTING.md#development-contributing-guidelines)
> When defining a method, enter a blank line before starting to write its body (except for methods with only one line of code, e.g. most getters and setters).

**Review by StefanKock — CHANGES_REQUESTED** [review]
Looks technically correct, I only have remarks how the code looks. When changing please amend the commit (easier to backport later).
I appreciate that the existing test was extended to prove that the bug is fixed.

**Comment by JonasCir** [comment]
@StefanKock Fixed the code style, removed the vars, and squashed everything into one commit.

**Review by StefanKock — APPROVED** [review]

**Review by FredrikSchaeferVitagroup — APPROVED** [review]

**Pull request merged**